### PR TITLE
Fix initial set of high/medium severity bugs found via Vijil audit

### PR DESCRIPTION
**Pull request by anuj-tambwekar**
Resolves 10 high-risk issues from the reliability scan (BC-1, BC-2, BC-3, BC-5, BC-6, BC-7, BC-14, BC-17, BC-18, BC-19).
                                                                                                                                                                     
  Trust Runtime Fail-Closed (BC-5, BC-6, BC-7):                                                                                                                      
  - BC-6: MAC fail-open on missing action — ToolPolicy.check now denies when allowed_actions is configured but the action key is missing, instead of silently permitting                                                                                                                                                         
  - BC-5: Silent guard degradation — enforce mode re-raises on Dome init failure; warn mode sets guards_disabled=True surfaced in EnforcementResult
  - BC-7: Silent mTLS downgrade — logs warning and emits structured audit event instead of bare except RuntimeError: pass                                            
                                                                                                                                                                     
  Guardrail Infrastructure (BC-2, BC-3, BC-17):                                                                                                                      
  - BC-17: PII in INFO-level logs — raw payloads removed from INFO; available at DEBUG behind VIJIL_LOG_PAYLOADS=1                                                   
  - BC-2: Sequential guard missing timeout — added asyncio.timeout(5) to sequential_guard and sequential_guard_batch                                                 
  - BC-3: ThreadPoolExecutor lifecycle — added close(), __del__, __enter__/__exit__ to Guardrail                    
                                                                                                                                                                     
  Embedding + Mode Validation (BC-14, BC-18):                         
  - BC-14: Class-level mutable lists — moved _items/_embeddings to __init__ so instances don't share state                                                           
  - BC-18: Mode validation — TrustRuntime rejects invalid mode strings; DomedMCPServer gains enforce parameter for shadow mode
                                                                                                                                                                     
  Error Visibility (BC-1, BC-19):                                                                                                                                    
  - BC-1: Detector error visibility — detectors returning label='error' now populate errored_methods on GuardResult, GuardrailResult, and ScanResult                 
  - BC-19: FAISS dimension validation — load_from_file() validates index dimension against embedder dimension                                                        
                                                                                                                                                                     
  Test plan                                                                                                                                                          
  
  - 4 new tests for BC-6 (missing/None action denial, valid action, no-regression)                                                                                   
  - 4 new tests for BC-5/BC-7 (enforce raises, warn degrades, guards_disabled surfaced, mTLS downgrade logged)
  - 2 new tests for BC-18 (invalid mode raises)                                                                                                                      
  - 2 new tests for BC-17 (INFO clean, DEBUG with env flag)           
  - 2 new tests for BC-2 (sequential timeout, batch timeout)                                                                                                         
  - 2 new tests for BC-3 (close shuts down executor, context manager)                                                                                                
  - 2 new tests for BC-14 (independent items, independent embeddings)                                                                                                
  - 3 new tests for BC-1 (errored surfaced, errored+triggered coexist, clean scan empty)                                                                             
  - 1 new test for BC-19 (dimension mismatch raises — skipped without faiss-cpu)                                                                                     
  - 146 passed, 0 regressions, ruff + mypy clean

### CODE REVIEW

**Review by ciphr — APPROVED** [review]
Approved.

**5 of 10 BCs cleanly closed** — BC-2 (sequential-guard timeout, both surfaces with substantive timeout-elapsed tests), BC-3 (Guardrail executor lifecycle: `close()` + `__del__` + ctx-manager, tests assert `_shutdown is True`), BC-6 (MAC fail-open on missing action, four tests covering missing / None / valid / no-`allowed_actions` paths), BC-14 (per-instance `_items` / `_embeddings` with `is not` assertions), BC-17 (PII at INFO de-leaked via the sync-→async transitive path), BC-19 (FAISS dim mismatch raises `ValueError`).

**4 BCs partially closed** plus a few new concerns the review surfaced. Filed as a single Phase-2 follow-up:

> [DOME-156 — Complete reliability sweep: finish BC-1 / BC-5 / BC-7 / BC-18 and address new concerns from PR #225](https://linear.app/vijil/issue/DOME-156)

Quick summary of what's in DOME-156:

| Item | Severity |
|---|---|
| BC-1 — Darwin OTel third-state (`detection.label` enum still only `"flagged" \| "clean"`; fleet error-rate observability still broken) | **High** |
| BC-1 — detector contract test for `label="error"` convention | Medium |
| BC-5 — structured audit event on warn-mode degrade (currently only `logger.warning`) | Medium |
| BC-7 — `identity.py:113-116` sibling mTLS surface untouched | Medium |
| BC-18 — `runtime.py:82` reintroduces the `model_copy(update={...})` post-validation bypass that BC-18 itself called out | Medium |
| 5s timeout as magic number at `guardrails/__init__.py:193,461` (CLAUDE.md §4) | Medium |
| `__del__` swallows all exceptions at `guardrails/__init__.py:592-595` | Low |
| Weak mtls downgrade test at `tests/trust/test_runtime.py:225-227` (swallows arbitrary exceptions) | Low |

The DOME-156 description has acceptance criteria + suggested fixes per item, and notes the two Low-severity items are deferrable.

Merging Phase 1 now keeps the 5 clean closures from sitting; Phase 2 (DOME-156) lands the remaining surfaces of the four partial BCs plus the magic-number cleanup.

File: vijil_dome/guardrails/__init__.py [hunk 1]
```diff
@@ -14,17 +14,20 @@
...
 import concurrent.futures
+import os
 from typing import List, Dict, Tuple, Union, Any, Optional  # noqa: F401
 from abc import ABC, abstractmethod
 import time
 from pydantic import BaseModel
 import logging
 
+_LOG_PAYLOADS = os.environ.get("VIJIL_LOG_PAYLOADS", "").strip() == "1"
+
```
[the remark this answers is not in the document]
Comment (anuj-tambwekar) [reply vijil_dome/guardrails/__init__.py]: Fixed

File: vijil_dome/trust/runtime.py [hunk 2]
```diff
@@ -348,8 +359,13 @@ def _verify_tool_identity(self, tool: Any) -> ToolAttestationStatus:
...
                     ctx.check_hostname = False
-                except RuntimeError:
-                    pass  # Fall back to server-only verification
+                except RuntimeError as exc:
+                    logger.warning("mTLS downgrade for %s: %s", tool.name, exc)
+                    self._audit._emit(
+                        "mtls_downgrade",
+                        tool_name=tool.name,
+                        error=str(exc),
+                    )
```
[the remark this answers is not in the document]
Comment (anuj-tambwekar) [reply vijil_dome/trust/runtime.py]: Fixed

File: vijil_dome/guardrails/__init__.py [hunk 3]
```diff
@@ -465,13 +510,18 @@ async def sequential_guard_batch(
...
+            ]
             items.append(GuardResult(
                 triggered=item_flagged[i],
                 details=item_detector_results[i],
                 exec_time=0.0,
                 response=item_response[i],
                 detection_score=detection_score,
                 triggered_methods=triggered_methods,
+                errored_methods=errored_methods,
             ))
```
[the remark this answers is not in the document]
Comment (anuj-tambwekar) [reply vijil_dome/guardrails/__init__.py:554]: Fixed

File: vijil_dome/guardrails/__init__.py [hunk 4]
```diff
@@ -185,13 +190,23 @@ async def sequential_guard(
...
+                        team_id=team_id,
+                        user_id=user_id,
+                    )
+            except TimeoutError:
+                logger.warning("Detection method %s timed out.", type(detector).__name__)
+                detector_scan_result = DetectionTimingResult(
+                    hit=False,
+                    result={"error": "Detection method timed out", "response_string": qs},
+                    exec_time=asyncio_timeout_limit * 1000,
+                )
```
[the remark this answers is not in the document]
Comment (anuj-tambwekar) [reply vijil_dome/guardrails/__init__.py]: Good callout. Fixed

File: vijil_dome/trust/runtime.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (anuj-tambwekar) [reply vijil_dome/trust/runtime.py:389]: This makes sense to me. I'll proceed by treating it as  a hard failure. If this becomes an issue later we can degrade it back to a warning

File: vijil_dome/guardrails/__init__.py [hunk 5]
```diff
@@ -578,13 +646,19 @@ async def sequential_guard(
...
+        ]
         return GuardrailResult(
             flagged=flagged,
             guardrail_response_message=response_string,
             exec_time=exec_time,
             guard_exec_details=guard_results,
             detection_score=detection_score,
             triggered_methods=triggered_methods,
+            errored_methods=errored_methods,
         )
```
[the remark this answers is not in the document]
Comment (anuj-tambwekar) [reply vijil_dome/guardrails/__init__.py:685]: Good catch

File: vijil_dome/trust/runtime.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (anuj-tambwekar) [reply vijil_dome/trust/runtime.py:389]: Done

File: vijil_dome/guardrails/__init__.py [hunk 6]
```diff
@@ -535,6 +585,24 @@ def __init__(
...
+        """Shut down the thread pool executor."""
+        if hasattr(self, "executor") and self.executor is not None:
+            self.executor.shutdown(wait=False, cancel_futures=True)
+
+    def __del__(self):
+        try:
+            self.close()
+        except Exception:
+            pass
+
```
[the remark this answers is not in the document]
Comment (anuj-tambwekar) [reply vijil_dome/guardrails/__init__.py]: Done

File: vijil_dome/tests/test_guardrail_infrastructure.py [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+    assert guardrail.executor._shutdown is True
+
+
+def test_guardrail_context_manager() -> None:
+    guard = Guard(guard_name="g", detector_list=[MockCleanDetector()])
+    with Guardrail(level="input", guard_list=[guard]) as guardrail:
+        assert guardrail.executor is not None
+        executor = guardrail.executor
+
+    assert executor._shutdown is True
```
[the remark this answers is not in the document]
Comment (anuj-tambwekar) [reply vijil_dome/tests/test_guardrail_infrastructure.py]: Done

File: vijil_dome/guardrails/__init__.py [hunk 8]
```diff
@@ -217,13 +232,20 @@ async def sequential_guard(
             default=0.0,
         )
         triggered_methods = [name for name, d in detector_results.items() if d.hit]
+        errored_methods = [
+            name for name, d in detector_results.items()
+            if not d.hit and isinstance(d.result, dict) and d.result.get("label") == "error"
+        ]
+        if errored_methods:
+            logger.warning("Detectors returned errors: %s", errored_methods)
```
[the remark this answers is not in the document]
Comment (anuj-tambwekar) [reply vijil_dome/guardrails/__init__.py:246]: Added. For this and the other errored methods, we're re-raising the exception to make sure that we're not listing methods that were cancelled via early exit as errors.

File: vijil_dome/guardrails/__init__.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (anuj-tambwekar) [reply vijil_dome/guardrails/__init__.py:208]: No this isn't a valid issue. Parallel guard calls have `asyncio.timeout(DETECTOR_TIMEOUT_SECONDS)` inside `detector_wrapper`. The guard runs detectors (each with a default 5s timeout), then does synchronous dict/list operations to compose the result. There's no I/O or await after the detector calls return. We don't want to throw an arbitrary timeout here either since the guards may be sequential or parallel. I believe this is a false positive
